graphbolt/src/partitioned_cache_policy.cc

```diff
@@ -0,0 +1,215 @@
+/**
+ *   Copyright (c) 2023, GT-TDAlab (Muhammed Fatih Balin & Umit V. Catalyurek)
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ * @file partitioned_cache_policy.cc
+ * @brief Partitioned cache policy implementation on the CPU.
+ */
+#include "./partitioned_cache_policy.h"
+
+#include <numeric>
+
+namespace graphbolt {
+namespace storage {
+
+constexpr int kIntGrainSize = 64;
+
+template <typename CachePolicy>
+PartitionedCachePolicy::PartitionedCachePolicy(
+    CachePolicy, int64_t capacity, int64_t num_partitions)
```

Comment:
why force user to pass in an instance of `CachePolicy` which is not used at all here?

Reply:
I had to instantiate the template somehow.